### PR TITLE
#comment: added gson parsing ktx

### LINKED ISSUE #9: Gson Extensions

### PR DESCRIPTION
**Pull request by ashoksinghal**

### CODE REVIEW

File: gson-ktx/gson-parsing-ktx.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+<h1 align="center">Awesome Android Extensions - <a href="https://chetangupta.net" target="_blank">AndroidBites</a>
+</h1>
+
+![awesome-android-extension-androidbites](./../androidbites_awesome_extension_banner.jpg)
```
Comment (ch8n) [inline gson-ktx/gson-parsing-ktx.md:4]: hi paths isn't correct can you please fix it

File: gson-ktx/gson-parsing-ktx.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    }
+    return null
+}
+```
+#### Usage
+```kotlin
+//test class used for demonstration
+ data class TestObject(val name: String?,
+                     val rollNo: Int?)
+
```
Comment (ch8n) [inline gson-ktx/gson-parsing-ktx.md:56]: please take care of formatting

File: gson-ktx/gson-parsing-ktx.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+* to Any type of object depending on the type provided]
+**/
+fun <T> String.fromJson(type: Type): T? {
+        if (this.isEmpty()) {
+            return null
+        }
+        try {
+            return gson.fromJson<T>(this, type)
+        } catch (exception: Exception) {
+        }
```
Comment (ch8n) [inline gson-ktx/gson-parsing-ktx.md]: please take care of formatting and indentations

File: gson-ktx/gson-parsing-ktx.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        return null
+    }
+```
+#### Usage
+```kotlin
+//test class used for demonstration
+data class TestObject(val name: String?,
+                      val rollNo: Int?)
+
+
```
Comment (ch8n) [inline gson-ktx/gson-parsing-ktx.md:86]: formatting please

**Review by ch8n — CHANGES_REQUESTED** [review]

File: gson-ktx/gson-parsing-ktx.md (hunk 1 above)
Comment (ashoksinghal) [reply gson-ktx/gson-parsing-ktx.md:4]: @ch8n its already the correct path.

File: gson-ktx/gson-parsing-ktx.md (hunk 2 above)
Comment (ashoksinghal) [reply gson-ktx/gson-parsing-ktx.md:56]: done

File: gson-ktx/gson-parsing-ktx.md (hunk 3 above)
Comment (ashoksinghal) [reply gson-ktx/gson-parsing-ktx.md]: done

File: gson-ktx/gson-parsing-ktx.md (hunk 4 above)
Comment (ashoksinghal) [reply gson-ktx/gson-parsing-ktx.md:86]: done

**Pull request merged**